### DISCUSSION TITLE
Motoko needs less parentheses (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4830] [Show and tell]
In many languages their grammar necessitates parentheses in many places to make them unambiguously parseable. In Motoko you can get away with much less. The rule of thumb is that if there is _that single thing_ in parentheses, then the parentheses are probably redundant. E.g. `#err e` instead of `#err(e)`, `if tracked {...}` instead of `if (tracked) {...}`. Same for `switch d`, `case null`, etc. You can even call functions in Bash-style: `foo arg` and define lambdas as `func p = ...` when the pattern is just a single token, though you have to consider whether your audience is okay with that!

**Comment by jessiemongeon1** [comment]
Scheduled for publication May 16th
